Sources/SCNPath/RKPathEntity.swift

```diff
@@ -0,0 +1,215 @@
+            loadShapes(scene: shapesScene)
+        } catch {
+            print(error.localizedDescription)
+            return
+        }
+        //Remove synchronization to save memory.
+        self.realityKitPath.visit(using: {$0.synchronization = nil})
+    }
+    
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    
+    
+    private func loadShapes(scene: ShapesProj.ShapesScene){
+        
+        //Change pivot point.
+        rectangleEntity = Entity()
+        rectangleEntity?.addChild(scene.rectangle!)
+        
+        circleEntity = scene.circle
+        
+        //Then we can use custom materials (even with transparency) and video materials.
+        //We could also smooth the points ourselves. They are all on the 2D X-Z plane.
+    }
+    
+    override func recalcGeometry() {
+        guard realityKitPath.isAnchored,
+              circleEntity != nil,
+              rectangleEntity != nil
+        else {return}
+        
+        //Start off with just a circle for the first position,
+        //Then every position after that gets a rectangle and a circle.
+        //This way, every rectangle has a circle on both ends.
+        if self.path.count == 1 {
+            let circleLocal = circleEntity!.clone(recursive: true)
+            realityKitPath.anchor!.addChild(circleLocal)
+            circleLocal.setPosition(self.path[0].getSIMD3Float(), relativeTo: nil)
+            pathEntitiesInScene.append(circleLocal)
+            return
+        }
+        
+        //Using a smaller distance between points on a curve will lead to smoother curves.
+        
+        //Keep all y-values the same so that the edges of the circles don't stick out.
+        let currentIndex = (path.count - 1)
+        let newPosition : SIMD3<Float> = [path[currentIndex].getSIMD3Float().x,
+                           path[0].y,
+                           path[currentIndex].getSIMD3Float().z]
+        let lastPosition : SIMD3<Float> = [path[currentIndex - 1].getSIMD3Float().x,
+                            path[0].y,
+                            path[currentIndex - 1].getSIMD3Float().z]
+        let length = simd_length(newPosition - lastPosition)
+        let rectangleLocal = rectangleEntity!.clone(recursive: true)
+        realityKitPath.anchor!.addChild(rectangleLocal)
+        pathEntitiesInScene.append(rectangleLocal)
+        rectangleLocal.position = lastPosition
+        rectangleLocal.position.y = newPosition.y
+
+        //The rectangle is already 1 meter long, so settings its scale in the Z-Dimension with the difference between the points (in meters) will leave it at the correct length, which is the distance between the points.
+        rectangleLocal.setScale([1, 1, length], relativeTo: nil)
+        //Rotate the rectangle to connect the dots.
+        // --(lastPosition is already at one end of the rectangle, rotate to put newPosition at the other end).
```

Comment:
this method is confusing me a little, is this for debugging?

Reply:
This was used to set the length of the segment. It was initially of length 1.0m on the z-dimension so scaling it in the z-direction will tell you the new length. My next implementation uses generated mesh instead of loading from a file so this scaling will no longer be needed.